### DISCUSSION TITLE
Getting api-ms-win-core-path-l1-1-0.dll is missing error

### DISCUSSION THREAD

**Discussion by BlohoJo** [FlareSolverr/FlareSolverr#793] [Q&A] [answered]
![error](https://github.com/FlareSolverr/FlareSolverr/assets/8242464/8a7bd735-960a-4f94-8fbf-e307f3a435fd)

I'm on Windows 7, so I'm guessing this is because FlareSolverr is Windows 10 only? :\

**Comment by ilike2burnthing** [comment]
32-bit: https://www.microsoft.com/en-us/download/details.aspx?id=47409

64-bit: https://www.microsoft.com/en-us/download/details.aspx?id=47442

**Comment by BlohoJo** [comment]
Whoops, I made a mistake and said I was on Windows 7, when it's actually Windows Server 2008 R2, which is based on Windows 7. I do that sometimes! :p

So my link should be https://www.microsoft.com/en-ph/download/details.aspx?id=47479, but it does download the same file as the link you posted (https://www.microsoft.com/en-us/download/details.aspx?id=47442). (Both downloaded files have the same filename and checksum.)

Unfortunately, it didn't work, I get the below message:

![error](https://github.com/FlareSolverr/FlareSolverr/assets/8242464/d352b20c-ee0c-4628-94f2-74c3f6451f2e)

`Windows update  could not be installed because of error 2149842967 "" (Command line: ""C:\Windows\system32\wusa.exe" "E:\USER\Install\Java\Windows6.1-KB3063858-x64.msu"        ")`

That update only appears for Windows Vista in the Microsoft Update Catalog: https://www.catalog.update.microsoft.com/Search.aspx?q=KB3063858

Not sure if there's a problem on my system, or if it's because that update was already installed.

![system](https://github.com/FlareSolverr/FlareSolverr/assets/8242464/43282b39-c4c0-41a2-a094-a8f06e7c18a2)

![system2](https://github.com/FlareSolverr/FlareSolverr/assets/8242464/4e74be6c-9030-41b9-a07e-4e2ae4ee4abf)

**Comment by BlohoJo** [comment]
I checked to see if the update was installed by running `wmic qfe list >winupdates.txt` and searching for KB3063858. It wasn't listed.

I force installed the update by extracting the .cab file from the .msu file. I then ran `DISM.exe /Online /Add-Package /PackagePath:c:\wip\KB3063858\Windows6.1-K3063858-x64.cab`. This worked:

```
Deployment Image Servicing and Management tool
Version: 6.1.7601.24499

Image Version: 6.1.7601.24499

Processing 1 of 1 - Adding package Package_for_KB3063858~31bf3856ad364e35~amd64~~6.1.1.0
[==========================100.0%==========================]
The operation completed successfully.
```

I checked that the update was installed by running `wmic qfe list >winupdates.txt`. The update is listed now:

`http://support.microsoft.com/?kbid=3063858     WIN-7DHRTQH0CV9  Security Update               KB3063858               WIN-7DHRTQH0CV9\Administrator  6/3/2023`

I rebooted after, but unfortunately, after all of that, I still get the same error when trying to run `flaresolverr.exe`:

![error](https://github.com/FlareSolverr/FlareSolverr/assets/8242464/8a7bd735-960a-4f94-8fbf-e307f3a435fd)

**Comment by ilike2burnthing** [comment]
https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170

**Comment by BlohoJo** [comment]
Uninstalled the below:

```
Microsoft Visual C++ 2015-2019 Redistributable (x64) 14.28.29910
Microsoft Visual C++ 2015-2019 Redistributable (x86) 14.28.29910
```

And installed https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170. Now I have:

```
Microsoft Visual C++ 2015-2022 Redistributable (x64) 14.36.32532
Microsoft Visual C++ 2015-2022 Redistributable (x86) 14.36.32532
```

I also have the Microsoft Visual C++ Redistributables for 2005, 2008, 2010, 2012, and 2013:

![mvc](https://github.com/FlareSolverr/FlareSolverr/assets/8242464/378b271b-374f-48dd-9d33-3aa8bd38c999)

Unfortunately again, it had no effect on the issue. The problem persists.

**Comment by BlohoJo** [comment]
I cleaned up the above, uninstalling all of the 2005-2013 redistributables, and installing the latest versions from https://www.techpowerup.com/download/visual-c-redistributable-runtime-package-all-in-one. I now have:

![mvc](https://github.com/FlareSolverr/FlareSolverr/assets/8242464/a1977ba7-580d-4155-8c6f-32a94ff5b63b)

Unfortunately again, it had no effect on the issue. The problem persists.

**Comment by BlohoJo** [comment, marked as answer]
I have discovered the answer:

```
flaresolverr\python311.dll
```

It uses Python 3.11. Windows 7 is not supported from Python 3.9 and above. It throws the error "api-ms-win-core-path-l1-1-0.dll is missing from your computer".
